control/src/main/java/org/kaazing/k3po/control/internal/Control.java

```diff
@@ -247,8 +281,11 @@ private PreparedEvent readPreparedEvent() throws IOException {
                 case "name":
                     // compatibility
                     break;
+                case "barrier":
+                    prepared.getBarriers().add(headerValue);
+                    break;
                 default:
-                    throw new IllegalStateException("Unrecognized event header: " + headerName);
+                    // NOP allow unrecognized headers for future compatibility
```

Comment:
Not sure how I would log here as this is on the TestFramework side of thing. (Needed to be logged via junit or System.out)